File: 1287_more_than_25%.js
var findSpecialInteger = function(arr) {    
    if(arr.length == 1){
        return arr[0]
    }
    let count = arr.length / 4;
    let hash = {};
    for(let i = 0; i < arr.length; i++){
        let ele = arr[i];
        if(!hash[ele]){
            hash[ele] = 1;
        } else if(hash[ele] + 1 > count){
            return ele;
        } else {
            hash[ele] += 1;
        }
    }

};

File: 1287_more_than_25%.py
def findSpecialInteger(self, arr: List[int]) -> int:
        amt = len(arr) / 4
        hash = {}
        for num in arr:
            if num in hash:
                hash[num] += 1
                if hash[num] >= amt:
                    return num
            else:
                hash[num] = 1